src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
```

Comment:
It's not so common to see the same type shared across tests, usually test case setup types are defined within the test itself. The reason this type is able to be defined at the top is because expectedResponse is an interface{}, however each test below uses its own defined expected response.

I'd move this into the test itself and change expectedResponse to a concrete type.

As a final nit, we've somewhat consistently chosen just "exp" for the expected value in test cases for the last year+ (though I'm sure exceptions can be found) -- I'd change expectedResponse to just exp.